src/lib/content/coordinators/GetContentItemV2Impl.ts

```diff
@@ -0,0 +1,67 @@
+import { GetContentItemById } from './GetContentItemById';
+import { GetContentItemByKey } from './GetContentItemByKey';
+import { ContentBody } from '../model/ContentBody';
+import { ContentItem } from '../model/ContentItem';
+import { createContentClient } from '../../client/createContentClient';
+import { ContentMapper } from '../mapper/ContentMapper';
+import { AxiosInstance, AxiosResponse } from 'axios';
+import { encodeQueryString } from '../../utils/Url';
+import { ContentClientConfigV2 } from '../../config/ContentClientConfigV2';
+
+/**
+ * @hidden
+ */
+export class GetContentItemV2Impl
+  implements GetContentItemById, GetContentItemByKey {
+  private readonly contentClient: AxiosInstance;
+
+  constructor(
+    private readonly config: ContentClientConfigV2,
+    private readonly mapper: ContentMapper
+  ) {
+    this.contentClient = createContentClient(
+      this.config,
+      `https://${config.hubName}.cdn.content.amplience.net`
+    );
+  }
+
+  getContentItemByKey<T extends ContentBody>(
+    key: string
+  ): Promise<ContentItem<T>> {
+    return this.getContentItem('key', key);
+  }
+  async getContentItemById<T extends ContentBody>(
+    id: string
+  ): Promise<ContentItem<T>> {
+    return this.getContentItem('id', id);
+  }
+
+  async getContentItem<T extends ContentBody>(
+    requestType: 'id' | 'key',
+    value: string
+  ): Promise<ContentItem<T>> {
+    const args = [
+      ['depth', 'all'],
+      ['format', 'inlined'],
+    ];
+    if (this.config.locale) {
+      args.push(['locale', this.config.locale]);
```

Comment:
we may need to support locale per request rather than having it purely config.

Reply:
Again I have followed the same pattern as Content Delivery 1, it currently doesn't expose this as a request option.

However I agree, but I suspect to implement this feature we will have to update the Content Delivery 1 implementation to also support locale per request.

Reply:
Happy to go with consistency over flexibility for now.